Unit test failure in tether

Seen in https://ci.vmware.run/vmware/vic/8347 (pasting the entire output in case the build is restarted):

```
Mar  2 2017 14:25:21.677Z INFO  Started test setup for github.com/vmware/vic/cmd/tether.attachCase
Mar  2 2017 14:25:21.678Z INFO  Using /tmp/tether.attachCase225869070 as test prefix
Mar  2 2017 14:25:21.678Z DEBUG created /tmp/tether.attachCase225869070/ttyS0{c,s} as raw conn pipes
Mar  2 2017 14:25:21.678Z DEBUG created /tmp/tether.attachCase225869070/ttyS1 as persistent log destinations
Mar  2 2017 14:25:21.679Z DEBUG created /tmp/tether.attachCase225869070/ttyS2 as persistent log destinations
Mar  2 2017 14:25:21.694Z DEBUG Test configuration: (func(string, string) error)(0x48b230)
Mar  2 2017 14:25:21.695Z INFO  Registering tether extension mocker
Mar  2 2017 14:25:21.695Z INFO  Registering tether extension Attach
Mar  2 2017 14:25:21.695Z INFO  opening ttyS0 pipe pair for backchannel (client)
Mar  2 2017 14:25:21.695Z DEBUG [BEGIN] [github.com/vmware/vic/lib/tether.(*tether).Start:365] main tether loop
Mar  2 2017 14:25:21.695Z DEBUG [BEGIN] [github.com/vmware/vic/lib/tether.(*tether).setup:111] main tether setup
Mar  2 2017 14:25:22.714Z INFO  Finished test teardown for github.com/vmware/vic/cmd/tether.attachCase
Mar  2 2017 14:25:22.714Z INFO  Started test setup for github.com/vmware/vic/cmd/tether.attachCase
Mar  2 2017 14:25:22.714Z INFO  Using /tmp/tether.attachCase844457237 as test prefix
Mar  2 2017 14:25:22.726Z INFO  Registering tether extension mocker
Mar  2 2017 14:25:22.727Z INFO  Registering tether extension Attach
Mar  2 2017 14:25:22.727Z INFO  opening ttyS0 pipe pair for backchannel (client)
Mar  2 2017 14:25:23.746Z INFO  Finished test teardown for github.com/vmware/vic/cmd/tether.attachCase
Mar  2 2017 14:25:23.746Z INFO  Started test setup for github.com/vmware/vic/cmd/tether.TestAttachMultiple
Mar  2 2017 14:25:23.746Z INFO  Using /tmp/tether.TestAttachMultiple954392432 as test prefix
Mar  2 2017 14:25:23.756Z INFO  Registering tether extension mocker
Mar  2 2017 14:25:23.757Z INFO  Registering tether extension Attach
Mar  2 2017 14:25:23.757Z INFO  opening ttyS0 pipe pair for backchannel (client)
Mar  2 2017 14:25:24.808Z DEBUG [ END ] [github.com/vmware/vic/lib/tether.(*tether).cleanup:167] [7.809114ms] main tether cleanup
Mar  2 2017 14:25:24.808Z DEBUG [BEGIN] [github.com/vmware/vic/pkg/serial.(*RawConn).Close:155]
Mar  2 2017 14:25:24.808Z DEBUG Closing the RawConn (/tmp/tether.TestAttachMultiple954392432/ttyS0c:/tmp/tether.TestAttachMultiple954392432/ttyS0s)
Mar  2 2017 14:25:24.808Z DEBUG Pushing EOF to any blocked readers on the raw connection (/tmp/tether.TestAttachMultiple954392432/ttyS0c:/tmp/tether.TestAttachMultiple954392432/ttyS0s)
Mar  2 2017 14:25:24.808Z DEBUG [ END ] [github.com/vmware/vic/pkg/serial.(*RawConn).Close:155] [28.436µs] 
Mar  2 2017 14:25:24.808Z DEBUG [BEGIN] [github.com/vmware/vic/pkg/serial.(*RawConn).Close:155]
Mar  2 2017 14:25:24.808Z DEBUG Close called again on RawConn (/tmp/tether.TestAttachMultiple954392432/ttyS0c:/tmp/tether.TestAttachMultiple954392432/ttyS0s) - dropping
Mar  2 2017 14:25:24.808Z DEBUG [ END ] [github.com/vmware/vic/pkg/serial.(*RawConn).Close:155] [5.665µs] 
Mar  2 2017 14:25:24.809Z INFO  Finished test teardown for github.com/vmware/vic/cmd/tether.TestAttachMultiple
Mar  2 2017 14:25:24.809Z INFO  Started test setup for github.com/vmware/vic/cmd/tether.TestAttachInvalid
Mar  2 2017 14:25:24.809Z INFO  Using /tmp/tether.TestAttachInvalid832270351 as test prefix
Mar  2 2017 14:25:24.809Z DEBUG created /tmp/tether.TestAttachInvalid832270351/ttyS0{c,s} as raw conn pipes
Mar  2 2017 14:25:24.809Z DEBUG created /tmp/tether.TestAttachInvalid832270351/ttyS1 as persistent log destinations
Mar  2 2017 14:25:24.809Z DEBUG created /tmp/tether.TestAttachInvalid832270351/ttyS2 as persistent log destinations
Mar  2 2017 14:25:24.811Z DEBUG Cleanup on detach from the session (non-tty)
Mar  2 2017 14:25:24.811Z DEBUG Removing [0xc420328490] from Errwriter
Mar  2 2017 14:25:24.811Z DEBUG [0xc4203d30b0] removing writer 0xc420328490 - currently 2 writers
Mar  2 2017 14:25:24.811Z DEBUG [0xc4203d30b0] removed writer - now 1 writers
Mar  2 2017 14:25:24.811Z DEBUG [0xc4203d30b0] Writer 0 [0xc4204340a0]
Mar  2 2017 14:25:24.811Z DEBUG Cleanup on detach from the session
Mar  2 2017 14:25:24.811Z DEBUG Removing [0xc42007a600] from Outwriter
Mar  2 2017 14:25:24.811Z DEBUG [0xc4203d3020] removing writer 0xc42007a600 - currently 2 writers
Mar  2 2017 14:25:24.811Z DEBUG [0xc4203d3020] removed writer - now 1 writers
Mar  2 2017 14:25:24.811Z DEBUG [0xc4203d3020] Writer 0 [0xc4204340a0]
Mar  2 2017 14:25:24.811Z DEBUG Removing [0xc42007a600] from Reader
Mar  2 2017 14:25:24.811Z DEBUG [0xc4203ef100] removing reader - currently 1 readers
Mar  2 2017 14:25:24.811Z DEBUG [0xc4203ef100] removed reader - now 0 readers
Mar  2 2017 14:25:24.811Z DEBUG [ END ] [github.com/vmware/vic/cmd/tether.(*attachServerSSH).channelMux:573] [18.42761ms] attach server channel request handler
Mar  2 2017 14:25:24.811Z DEBUG [0xc4203ef100] removing reader due to EOF
Mar  2 2017 14:25:24.811Z DEBUG [0xc4203ef100] removing reader - currently 0 readers
Mar  2 2017 14:25:24.811Z DEBUG [0xc4203ef100] read "" from 1 readers (err: <nil>)
Mar  2 2017 14:25:24.811Z DEBUG [0xc4203ef100] read from closed multi-reader, returning EOF
Mar  2 2017 14:25:24.811Z DEBUG [0xc4203ef100] read "" from 0 readers (err: &errors.errorString{s:"EOF"})
Mar  2 2017 14:25:24.811Z DEBUG Reader stdin returned: %!s(<nil>)
Mar  2 2017 14:25:24.811Z DEBUG [BEGIN] [github.com/vmware/vic/pkg/serial.(*RawConn).Close:155]
Mar  2 2017 14:25:24.811Z DEBUG Close called again on RawConn (/tmp/tether.TestAttachMultiple954392432/ttyS0s:/tmp/tether.TestAttachMultiple954392432/ttyS0c) - dropping
Mar  2 2017 14:25:24.811Z DEBUG [ END ] [github.com/vmware/vic/pkg/serial.(*RawConn).Close:155] [8.242µs] 
Mar  2 2017 14:25:24.811Z INFO  Stopped reaping child processes
Mar  2 2017 14:25:24.811Z DEBUG Returning error from read: EOF
Mar  2 2017 14:25:24.811Z DEBUG [ END ] [github.com/vmware/vic/pkg/serial.(*RawConn).Read:100] [10.978067ms] 
Mar  2 2017 14:25:24.811Z DEBUG [BEGIN] [github.com/vmware/vic/pkg/serial.(*RawConn).Close:155]
Mar  2 2017 14:25:24.811Z DEBUG Close called again on RawConn (/tmp/tether.TestAttachMultiple954392432/ttyS0s:/tmp/tether.TestAttachMultiple954392432/ttyS0c) - dropping
Mar  2 2017 14:25:24.811Z DEBUG [ END ] [github.com/vmware/vic/pkg/serial.(*RawConn).Close:155] [6.539µs] 
Mar  2 2017 14:25:24.811Z DEBUG [ END ] [github.com/vmware/vic/cmd/tether.(*attachServerSSH).globalMux:507] [27.745153ms] attach server global request handler
Mar  2 2017 14:25:24.811Z INFO  Incoming attach channel closed
Mar  2 2017 14:25:24.811Z DEBUG cleanup on connection
Mar  2 2017 14:25:24.819Z DEBUG Returning error from read: EOF
Mar  2 2017 14:25:24.819Z DEBUG [ END ] [github.com/vmware/vic/pkg/serial.(*RawConn).Read:100] [18.606286ms] 
Mar  2 2017 14:25:24.819Z DEBUG [BEGIN] [github.com/vmware/vic/pkg/serial.(*RawConn).Close:155]
Mar  2 2017 14:25:24.819Z DEBUG Close called again on RawConn (/tmp/tether.TestAttachMultiple954392432/ttyS0c:/tmp/tether.TestAttachMultiple954392432/ttyS0s) - dropping
Mar  2 2017 14:25:24.819Z DEBUG [ END ] [github.com/vmware/vic/pkg/serial.(*RawConn).Close:155] [12.842µs] 
Mar  2 2017 14:25:24.819Z DEBUG Writer goroutine for stdout returned: %!s(<nil>)
Mar  2 2017 14:25:24.819Z DEBUG (stdout): Writing zero byte to stdin pipe
Mar  2 2017 14:25:24.819Z DEBUG (stdout): Closing stdin pipe
Mar  2 2017 14:25:24.819Z DEBUG Writer goroutine for stdout exiting
Mar  2 2017 14:25:24.819Z DEBUG Writer goroutine for stderr returned: %!s(<nil>)
Mar  2 2017 14:25:24.819Z DEBUG (stderr): Writing zero byte to stdin pipe
Mar  2 2017 14:25:24.819Z DEBUG (stderr): Closing stdin pipe
Mar  2 2017 14:25:24.819Z DEBUG Writer goroutine for stderr exiting
Mar  2 2017 14:25:24.811Z DEBUG [ END ] [github.com/vmware/vic/cmd/tether.(*attachServerSSH).run:303] [1.049755722s] main attach server loop
Mar  2 2017 14:25:24.821Z DEBUG Test configuration: (func(string, string) error)(0x48b230)
Mar  2 2017 14:25:24.821Z INFO  Registering tether extension mocker
Mar  2 2017 14:25:24.821Z INFO  Registering tether extension Attach
Mar  2 2017 14:25:24.821Z INFO  opening ttyS0 pipe pair for backchannel (client)
Mar  2 2017 14:25:24.825Z DEBUG [BEGIN] [github.com/vmware/vic/lib/tether.(*tether).Start:365] main tether loop
Mar  2 2017 14:25:24.825Z DEBUG [BEGIN] [github.com/vmware/vic/lib/tether.(*tether).setup:111] main tether setup
Mar  2 2017 14:25:25.827Z INFO  Stopped reaping child processes
Mar  2 2017 14:25:25.827Z INFO  Finished test teardown for github.com/vmware/vic/cmd/tether.TestAttachInvalid
Mar  2 2017 14:25:25.827Z INFO  Started test setup for github.com/vmware/vic/cmd/tether.TestReattach
Mar  2 2017 14:25:25.827Z INFO  Using /tmp/tether.TestReattach708512290 as test prefix
Mar  2 2017 14:25:25.830Z INFO  Registering tether extension mocker
Mar  2 2017 14:25:25.830Z INFO  Registering tether extension Attach
Mar  2 2017 14:25:25.830Z INFO  opening ttyS0 pipe pair for backchannel (client)
SIGQUIT: quit
PC=0x4613e9 m=0

goroutine 0 [idle]:
runtime.epollwait(0xf, 0x7ffdf5b4ad18, 0xffffffff00000080, 0x0, 0xffffffff00000000, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/local/go/src/runtime/sys_linux_amd64.s:525 +0x19
runtime.netpoll(0x15e7801, 0x0)
	/usr/local/go/src/runtime/netpoll_epoll.go:67 +0x93
runtime.findrunnable(0xc42001f500, 0x0)
	/usr/local/go/src/runtime/proc.go:2003 +0x28c
runtime.schedule()
	/usr/local/go/src/runtime/proc.go:2120 +0x14c
runtime.park_m(0xc4204141a0)
	/usr/local/go/src/runtime/proc.go:2183 +0x123
runtime.mcall(0x7ffdf5b4b490)
	/usr/local/go/src/runtime/asm_amd64.s:240 +0x5b

goroutine 1 [chan receive, 9 minutes]:
testing.(*T).Run(0xc42007a0c0, 0xda5548, 0xc, 0xe0b348, 0xc42004bc01)
	/usr/local/go/src/testing/testing.go:647 +0x316
testing.RunTests.func1(0xc42007a0c0)
	/usr/local/go/src/testing/testing.go:793 +0x6d
testing.tRunner(0xc42007a0c0, 0xc42004bcf8)
	/usr/local/go/src/testing/testing.go:610 +0x81
testing.RunTests(0xe0b6e8, 0x15bab20, 0x6, 0x6, 0xd9f9fe)
	/usr/local/go/src/testing/testing.go:799 +0x319
testing.(*M).Run(0xc42004bee8, 0x4)
	/usr/local/go/src/testing/testing.go:743 +0x85
github.com/vmware/vic/cmd/tether.TestMain(0xc42004bee8)
	/drone/src/github.com/vmware/vic/cmd/tether/tether_test.go:172 +0x139
main.main()
	github.com/vmware/vic/cmd/tether/_test/_testmain.go:116 +0x1bd

goroutine 17 [syscall, 9 minutes, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2086 +0x1

goroutine 18 [select, 9 minutes, locked to thread]:
runtime.gopark(0xe0bf50, 0x0, 0xda04ae, 0x6, 0x18, 0x2)
	/usr/local/go/src/runtime/proc.go:259 +0x13a
runtime.selectgoImpl(0xc42003bf20, 0x0, 0x18)
	/usr/local/go/src/runtime/select.go:423 +0x1235
runtime.selectgo(0xc42003bf20)
	/usr/local/go/src/runtime/select.go:238 +0x1c
runtime.ensureSigM.func1()
	/usr/local/go/src/runtime/signal1_unix.go:304 +0x2f3
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2086 +0x1

goroutine 6 [syscall, 9 minutes]:
os/signal.signal_recv(0x15c72e0)
	/usr/local/go/src/runtime/sigqueue.go:116 +0x157
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x22
created by os/signal.init.1
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x41

goroutine 340 [chan receive, 9 minutes]:
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*mux).openChannel(0xc4204cbab0, 0xd9fb72, 0x6, 0xc420376548, 0x6, 0x8, 0x6, 0x8, 0xc420376548)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/mux.go:322 +0x204
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*mux).OpenChannel(0xc4204cbab0, 0xd9fb72, 0x6, 0xc420376548, 0x6, 0x8, 0x0, 0x0, 0x0, 0x0, ...)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/mux.go:298 +0x64
github.com/vmware/vic/lib/portlayer/attach.SSHAttach(0xc4204f4cc0, 0xd9fb72, 0x6, 0x0, 0x0, 0x0, 0x0)
	/drone/src/github.com/vmware/vic/lib/portlayer/attach/client.go:84 +0x16d
github.com/vmware/vic/cmd/tether.TestReattach.func2()
	/drone/src/github.com/vmware/vic/cmd/tether/attach_test.go:795 +0x14d
github.com/vmware/vic/cmd/tether.TestReattach(0xc42007a900)
	/drone/src/github.com/vmware/vic/cmd/tether/attach_test.go:821 +0x808
testing.tRunner(0xc42007a900, 0xe0b348)
	/usr/local/go/src/testing/testing.go:610 +0x81
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:646 +0x2ec

goroutine 284 [syscall, 9 minutes]:
syscall.Syscall(0x0, 0x14, 0xc42023e000, 0x3800, 0xc4200dfd38, 0x5518dd, 0xc42007a3c0)
	/usr/local/go/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.read(0x14, 0xc42023e000, 0x3800, 0x3800, 0x4a0f9c, 0x40deef, 0x15ec4a0)
	/usr/local/go/src/syscall/zsyscall_linux_amd64.go:783 +0x55
syscall.Read(0x14, 0xc42023e000, 0x3800, 0x3800, 0x4a2fd5, 0xc420326ae0, 0x1)
	/usr/local/go/src/syscall/syscall_unix.go:161 +0x49
os.(*File).read(0xc4202bc120, 0xc42023e000, 0x3800, 0x3800, 0x1000000000000, 0xc4200dfe08, 0x40f859)
	/usr/local/go/src/os/file_unix.go:228 +0x4f
os.(*File).Read(0xc4202bc120, 0xc42023e000, 0x3800, 0x3800, 0x0, 0x40deef, 0x15eb3f8)
	/usr/local/go/src/os/file.go:101 +0x59
io.copyBuffer(0x7fb6b8d0c318, 0xc42023a8a0, 0x15c30a0, 0xc4202bc120, 0xc42023e000, 0x3800, 0x3800, 0x40f52f, 0xbfd680, 0xd865e0)
	/usr/local/go/src/io/io.go:390 +0x147
io.CopyBuffer(0x7fb6b8d0c318, 0xc42023a8a0, 0x15c30a0, 0xc4202bc120, 0xc42023e000, 0x3800, 0x3800, 0x7fb6b8d0c3e0, 0xc42007a3c0, 0xd)
	/usr/local/go/src/io/io.go:371 +0x88
github.com/vmware/vic/lib/tether.establishNonPty.func3(0xc42030e580)
	/drone/src/github.com/vmware/vic/lib/tether/tether_linux.go:306 +0x135
created by github.com/vmware/vic/lib/tether.establishNonPty
	/drone/src/github.com/vmware/vic/lib/tether/tether_linux.go:322 +0x310

goroutine 400 [chan receive, 9 minutes]:
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*Client).handleGlobalRequests(0xc4204f4640, 0xc4204fc120)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/client.go:132 +0x4e
created by github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.NewClient
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/client.go:55 +0xee

goroutine 433 [chan receive, 9 minutes]:
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*Client).handleChannelOpens(0xc4204f4cc0, 0xc4204fc060)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/client.go:141 +0x73
created by github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.NewClient
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/client.go:56 +0x11a

goroutine 272 [chan send, 9 minutes]:
github.com/vmware/vic/cmd/tether.(*attachServerSSH).run(0xc42027a0e0, 0x0, 0x0)
	github.com/vmware/vic/cmd/tether/_test/_obj_test/attach.go:494 +0xdeb
created by github.com/vmware/vic/cmd/tether.(*attachServerSSH).start
	github.com/vmware/vic/cmd/tether/_test/_obj_test/attach.go:199 +0x3c0

goroutine 435 [chan receive, 9 minutes]:
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*forwardList).handleChannels(0xc4204f4cd0, 0xc4204e0c00)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/tcpip.go:178 +0x71
created by github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.NewClient
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/client.go:61 +0x18f

goroutine 282 [semacquire, 9 minutes]:
sync.runtime_notifyListWait(0xc4202343d0, 0xc400000000)
	/usr/local/go/src/runtime/sema.go:267 +0x122
sync.(*Cond).Wait(0xc4202343c0)
	/usr/local/go/src/sync/cond.go:57 +0x80
github.com/vmware/vic/pkg/dio.(*multiReader).Read(0xc420234380, 0xc420242000, 0x3800, 0x3800, 0x0, 0x0, 0x0)
	/drone/src/github.com/vmware/vic/pkg/dio/reader.go:71 +0x25d
io.copyBuffer(0x7fb6b8d0c278, 0xc42023aba0, 0x7fb6b8d0c2a0, 0xc420234380, 0xc420242000, 0x3800, 0x3800, 0x40f52f, 0xbfd680, 0xc6b2c0)
	/usr/local/go/src/io/io.go:390 +0x147
io.CopyBuffer(0x7fb6b8d0c278, 0xc42023aba0, 0x7fb6b8d0c2a0, 0xc420234380, 0xc420242000, 0x3800, 0x3800, 0xc42016ad80, 0xc42030e580, 0x0)
	/usr/local/go/src/io/io.go:371 +0x88
github.com/vmware/vic/lib/tether.establishNonPty.func1(0xc42030e580)
	/drone/src/github.com/vmware/vic/lib/tether/tether_linux.go:273 +0x12d
created by github.com/vmware/vic/lib/tether.establishNonPty
	/drone/src/github.com/vmware/vic/lib/tether/tether_linux.go:281 +0x36b

goroutine 440 [syscall, 9 minutes]:
syscall.Syscall(0x0, 0xc, 0xc42047b000, 0x1000, 0x456de0, 0xc4202c1a00, 0x4)
	/usr/local/go/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.read(0xc, 0xc42047b000, 0x1000, 0x1000, 0xc4204f2930, 0x40fe9c, 0xc4204f2988)
	/usr/local/go/src/syscall/zsyscall_linux_amd64.go:783 +0x55
syscall.Read(0xc, 0xc42047b000, 0x1000, 0x1000, 0xc4200e1728, 0x4065bb, 0xc4204f2988)
	/usr/local/go/src/syscall/syscall_unix.go:161 +0x49
os.(*File).read(0xc42002a0e8, 0xc42047b000, 0x1000, 0x1000, 0x0, 0x0, 0xc4204f2930)
	/usr/local/go/src/os/file_unix.go:228 +0x4f
os.(*File).Read(0xc42002a0e8, 0xc42047b000, 0x1000, 0x1000, 0x50, 0x0, 0x434998)
	/usr/local/go/src/os/file.go:101 +0x59
github.com/vmware/vic/pkg/serial.(*RawConn).Read.func2(0xc420376590, 0xc4203765a0, 0xc42016bc80, 0xc42047b000, 0x1000, 0x1000, 0xc4204f2a10)
	/drone/src/github.com/vmware/vic/pkg/serial/rawconn.go:117 +0x55
created by github.com/vmware/vic/pkg/serial.(*RawConn).Read
	/drone/src/github.com/vmware/vic/pkg/serial/rawconn.go:123 +0x152

goroutine 185 [semacquire, 9 minutes]:
sync.runtime_notifyListWait(0xc4203ef1d0, 0xc400000000)
	/usr/local/go/src/runtime/sema.go:267 +0x122
sync.(*Cond).Wait(0xc4203ef1c0)
	/usr/local/go/src/sync/cond.go:57 +0x80
github.com/vmware/vic/pkg/dio.(*multiReader).Read(0xc4203ef180, 0xc4202f8000, 0x3800, 0x3800, 0x0, 0x0, 0x0)
	/drone/src/github.com/vmware/vic/pkg/dio/reader.go:71 +0x25d
io.copyBuffer(0x7fb6b8d0c278, 0xc4203d3800, 0x7fb6b8d0c2a0, 0xc4203ef180, 0xc4202f8000, 0x3800, 0x3800, 0x40f52f, 0xbfd680, 0xc6b2c0)
	/usr/local/go/src/io/io.go:390 +0x147
io.CopyBuffer(0x7fb6b8d0c278, 0xc4203d3800, 0x7fb6b8d0c2a0, 0xc4203ef180, 0xc4202f8000, 0x3800, 0x3800, 0x1, 0x1, 0xc42033efa0)
	/usr/local/go/src/io/io.go:371 +0x88
github.com/vmware/vic/lib/tether.establishNonPty.func1(0xc4202ce580)
	/drone/src/github.com/vmware/vic/lib/tether/tether_linux.go:273 +0x12d
created by github.com/vmware/vic/lib/tether.establishNonPty
	/drone/src/github.com/vmware/vic/lib/tether/tether_linux.go:281 +0x36b

goroutine 187 [syscall, 9 minutes]:
syscall.Syscall(0x0, 0xb, 0xc42040a000, 0x3800, 0x7fb6b8d60000, 0x0, 0x0)
	/usr/local/go/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.read(0xb, 0xc42040a000, 0x3800, 0x3800, 0xc420286e10, 0x40deef, 0x15ec4a0)
	/usr/local/go/src/syscall/zsyscall_linux_amd64.go:783 +0x55
syscall.Read(0xb, 0xc42040a000, 0x3800, 0x3800, 0xc4202c1380, 0xc42033f5d0, 0x1)
	/usr/local/go/src/syscall/syscall_unix.go:161 +0x49
os.(*File).read(0xc4202bc038, 0xc42040a000, 0x3800, 0x3800, 0x100000040fe9c, 0xc42033f608, 0x40f859)
	/usr/local/go/src/os/file_unix.go:228 +0x4f
os.(*File).Read(0xc4202bc038, 0xc42040a000, 0x3800, 0x3800, 0x0, 0x40deef, 0x15eb3f8)
	/usr/local/go/src/os/file.go:101 +0x59
io.copyBuffer(0x7fb6b8d0c318, 0xc4203d3230, 0x15c30a0, 0xc4202bc038, 0xc42040a000, 0x3800, 0x3800, 0x40f52f, 0xbfd680, 0xd865e0)
	/usr/local/go/src/io/io.go:390 +0x147
io.CopyBuffer(0x7fb6b8d0c318, 0xc4203d3230, 0x15c30a0, 0xc4202bc038, 0xc42040a000, 0x3800, 0x3800, 0x0, 0xbd4780, 0xc4202943c0)
	/usr/local/go/src/io/io.go:371 +0x88
github.com/vmware/vic/lib/tether.establishNonPty.func3(0xc4202ce580)
	/drone/src/github.com/vmware/vic/lib/tether/tether_linux.go:306 +0x135
created by github.com/vmware/vic/lib/tether.establishNonPty
	/drone/src/github.com/vmware/vic/lib/tether/tether_linux.go:322 +0x310

goroutine 354 [chan receive, 9 minutes]:
github.com/vmware/vic/cmd/tether.backchannel.func1(0x15c88a0, 0xc42026a0c0, 0x15caec0, 0xc42016bc80)
	github.com/vmware/vic/cmd/tether/_test/_obj_test/attach.go:255 +0x6b
created by github.com/vmware/vic/cmd/tether.backchannel
	github.com/vmware/vic/cmd/tether/_test/_obj_test/attach.go:259 +0x119

goroutine 186 [syscall, 9 minutes]:
syscall.Syscall(0x0, 0x9, 0xc420492000, 0x3800, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.read(0x9, 0xc420492000, 0x3800, 0x3800, 0x0, 0x40deef, 0x15ec4a0)
	/usr/local/go/src/syscall/zsyscall_linux_amd64.go:783 +0x55
syscall.Read(0x9, 0xc420492000, 0x3800, 0x3800, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/syscall_unix.go:161 +0x49
os.(*File).read(0xc4202bc028, 0xc420492000, 0x3800, 0x3800, 0x4566e0, 0xc420340608, 0x40f859)
	/usr/local/go/src/os/file_unix.go:228 +0x4f
os.(*File).Read(0xc4202bc028, 0xc420492000, 0x3800, 0x3800, 0x0, 0x40deef, 0x15eb3f8)
	/usr/local/go/src/os/file.go:101 +0x59
io.copyBuffer(0x7fb6b8d0c318, 0xc4203d31a0, 0x15c30a0, 0xc4202bc028, 0xc420492000, 0x3800, 0x3800, 0x40f52f, 0xbfd680, 0xd865e0)
	/usr/local/go/src/io/io.go:390 +0x147
io.CopyBuffer(0x7fb6b8d0c318, 0xc4203d31a0, 0x15c30a0, 0xc4202bc028, 0xc420492000, 0x3800, 0x3800, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:371 +0x88
github.com/vmware/vic/lib/tether.establishNonPty.func2(0xc4202ce580)
	/drone/src/github.com/vmware/vic/lib/tether/tether_linux.go:287 +0x135
created by github.com/vmware/vic/lib/tether.establishNonPty
	/drone/src/github.com/vmware/vic/lib/tether/tether_linux.go:303 +0x2eb

goroutine 387 [select, 9 minutes]:
github.com/vmware/vic/pkg/serial.(*RawConn).Read(0xc42016bc80, 0xc42047b000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/drone/src/github.com/vmware/vic/pkg/serial/rawconn.go:129 +0x42e
bufio.(*Reader).fill(0xc4204e01e0)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Read(0xc4204e01e0, 0xc4204714a0, 0x5, 0x5, 0x0, 0x0, 0x3000000000000)
	/usr/local/go/src/bufio/bufio.go:209 +0x1bc
io.ReadAtLeast(0x15c1ae0, 0xc4204e01e0, 0xc4204714a0, 0x5, 0x5, 0x5, 0x20, 0x100000000000020, 0x0)
	/usr/local/go/src/io/io.go:307 +0xa4
io.ReadFull(0x15c1ae0, 0xc4204e01e0, 0xc4204714a0, 0x5, 0x5, 0x30, 0x1, 0x0)
	/usr/local/go/src/io/io.go:325 +0x58
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*streamPacketCipher).readPacket(0xc420471480, 0xc400000011, 0x15c1ae0, 0xc4204e01e0, 0x1d, 0x1d, 0x410099, 0xc4204e0358, 0x42f1c4)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/cipher.go:149 +0x80
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*connectionState).readPacket(0xc4202b9300, 0xc4204e01e0, 0xc4204f1560, 0x0, 0x1d, 0x0, 0x0)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/transport.go:93 +0x68
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*transport).readPacket(0xc4202b9300, 0x455120, 0xc4204e0300, 0x1d, 0x0, 0xc420449f28)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/transport.go:89 +0x39
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*handshakeTransport).readOnePacket(0xc42031d560, 0xc4204e0300, 0xc420449f70, 0x1d, 0x0, 0x0)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/handshake.go:157 +0x4f
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*handshakeTransport).readLoop(0xc42031d560)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/handshake.go:129 +0x2f
created by github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.newServerTransport
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/handshake.go:104 +0x216

goroutine 180 [IO wait, 9 minutes]:
net.runtime_pollWait(0x7fb6b8ccc200, 0x72, 0x0)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc420019250, 0x72, 0xc42003dd28, 0xc4200140e8)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc420019250, 0x15c47e0, 0xc4200140e8)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).accept(0xc4200191f0, 0x0, 0x15c31a0, 0xc420407e80)
	/usr/local/go/src/net/fd_unix.go:419 +0x238
net.(*TCPListener).accept(0xc42002a028, 0xc420111890, 0xc42003de38, 0x4fecf0)
	/usr/local/go/src/net/tcpsock_posix.go:132 +0x2e
net.(*TCPListener).AcceptTCP(0xc42002a028, 0xc4202a22dc, 0xc42003de28, 0x713b20)
	/usr/local/go/src/net/tcpsock.go:209 +0x49
net/http.tcpKeepAliveListener.Accept(0xc42002a028, 0xc420111860, 0xbcde20, 0x15b4390, 0xcd1e20)
	/usr/local/go/src/net/http/server.go:2608 +0x2f
net/http.(*Server).Serve(0xc4202a2280, 0x15c82a0, 0xc42002a028, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:2273 +0x1ce
net/http.(*Server).ListenAndServe(0xc4202a2280, 0xc4202a2280, 0x0)
	/usr/local/go/src/net/http/server.go:2219 +0xb4
net/http.ListenAndServe(0xda4d38, 0xc, 0x0, 0x0, 0xc4201edd60, 0xc4201edd70)
	/usr/local/go/src/net/http/server.go:2351 +0xa0
created by github.com/vmware/vic/lib/tether.(*tether).setLogLevel.func1
	/drone/src/github.com/vmware/vic/lib/tether/tether.go:207 +0x5c

goroutine 271 [chan receive, 9 minutes]:
github.com/vmware/vic/lib/tether.(*tether).childReaper.func1(0xc42025b740)
	/drone/src/github.com/vmware/vic/lib/tether/tether_linux.go:78 +0x68
created by github.com/vmware/vic/lib/tether.(*tether).childReaper
	/drone/src/github.com/vmware/vic/lib/tether/tether_linux.go:134 +0x1e3

goroutine 283 [syscall, 9 minutes]:
syscall.Syscall(0x0, 0x10, 0xc420246000, 0x3800, 0xc42033fd38, 0xc42033fd48, 0x456bd0)
	/usr/local/go/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.read(0x10, 0xc420246000, 0x3800, 0x3800, 0xc4200ee5a0, 0x40deef, 0x15ec4a0)
	/usr/local/go/src/syscall/zsyscall_linux_amd64.go:783 +0x55
syscall.Read(0x10, 0xc420246000, 0x3800, 0x3800, 0x4cb093, 0x15b4980, 0x1)
	/usr/local/go/src/syscall/syscall_unix.go:161 +0x49
os.(*File).read(0xc4202bc110, 0xc420246000, 0x3800, 0x3800, 0x1000000000000, 0xc42033fe08, 0x40f859)
	/usr/local/go/src/os/file_unix.go:228 +0x4f
os.(*File).Read(0xc4202bc110, 0xc420246000, 0x3800, 0x3800, 0x0, 0x40deef, 0x15eb3f8)
	/usr/local/go/src/os/file.go:101 +0x59
io.copyBuffer(0x7fb6b8d0c318, 0xc42023a870, 0x15c30a0, 0xc4202bc110, 0xc420246000, 0x3800, 0x3800, 0x40f52f, 0xbfd680, 0xd865e0)
	/usr/local/go/src/io/io.go:390 +0x147
io.CopyBuffer(0x7fb6b8d0c318, 0xc42023a870, 0x15c30a0, 0xc4202bc110, 0xc420246000, 0x3800, 0x3800, 0xc42033ffa0, 0x1, 0x1)
	/usr/local/go/src/io/io.go:371 +0x88
github.com/vmware/vic/lib/tether.establishNonPty.func2(0xc42030e580)
	/drone/src/github.com/vmware/vic/lib/tether/tether_linux.go:287 +0x135
created by github.com/vmware/vic/lib/tether.establishNonPty
	/drone/src/github.com/vmware/vic/lib/tether/tether_linux.go:303 +0x2eb

goroutine 401 [chan receive, 9 minutes]:
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*Client).handleChannelOpens(0xc4204f4640, 0xc4204fc060)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/client.go:141 +0x73
created by github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.NewClient
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/client.go:56 +0x11a

goroutine 341 [chan receive, 9 minutes]:
github.com/vmware/vic/lib/tether.(*tether).Start(0xc42025b740, 0x0, 0x0)
	/drone/src/github.com/vmware/vic/lib/tether/tether.go:373 +0x141
github.com/vmware/vic/cmd/tether.StartAttachTether.func1(0xc42007a900)
	/drone/src/github.com/vmware/vic/cmd/tether/tether_test.go:191 +0x39
created by github.com/vmware/vic/cmd/tether.StartAttachTether
	/drone/src/github.com/vmware/vic/cmd/tether/tether_test.go:195 +0x2f8

goroutine 418 [semacquire, 9 minutes]:
sync.runtime_notifyListWait(0xc4204ce990, 0x1)
	/usr/local/go/src/runtime/sema.go:267 +0x122
sync.(*Cond).Wait(0xc4204ce980)
	/usr/local/go/src/sync/cond.go:57 +0x80
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*mux).Wait(0xc4204cbab0, 0x0, 0x0)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/mux.go:110 +0x9c
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.NewClient.func1(0xc4204f4640)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/client.go:58 +0x33
created by github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.NewClient
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/client.go:60 +0x13c

goroutine 432 [chan receive, 9 minutes]:
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*Client).handleGlobalRequests(0xc4204f4cc0, 0xc4204fc120)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/client.go:132 +0x4e
created by github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.NewClient
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/client.go:55 +0xee

goroutine 434 [semacquire, 9 minutes]:
sync.runtime_notifyListWait(0xc4204ce990, 0x2)
	/usr/local/go/src/runtime/sema.go:267 +0x122
sync.(*Cond).Wait(0xc4204ce980)
	/usr/local/go/src/sync/cond.go:57 +0x80
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*mux).Wait(0xc4204cbab0, 0x0, 0x0)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/mux.go:110 +0x9c
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.NewClient.func1(0xc4204f4cc0)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/client.go:58 +0x33
created by github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.NewClient
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/client.go:60 +0x13c

goroutine 419 [chan receive, 9 minutes]:
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*forwardList).handleChannels(0xc4204f4650, 0xc4204e0960)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/tcpip.go:178 +0x71
created by github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.NewClient
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/client.go:61 +0x18f

goroutine 375 [select, 9 minutes]:
github.com/vmware/vic/pkg/serial.(*RawConn).Read(0xc42025b980, 0xc4204e7000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/drone/src/github.com/vmware/vic/pkg/serial/rawconn.go:129 +0x42e
bufio.(*Reader).fill(0xc42025ba40)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Read(0xc42025ba40, 0xc420255920, 0x5, 0x5, 0x0, 0x0, 0x3000000000000)
	/usr/local/go/src/bufio/bufio.go:209 +0x1bc
io.ReadAtLeast(0x15c1ae0, 0xc42025ba40, 0xc420255920, 0x5, 0x5, 0x5, 0x20, 0x100000000000020, 0x20)
	/usr/local/go/src/io/io.go:307 +0xa4
io.ReadFull(0x15c1ae0, 0xc42025ba40, 0xc420255920, 0x5, 0x5, 0x40, 0x1, 0x0)
	/usr/local/go/src/io/io.go:325 +0x58
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*streamPacketCipher).readPacket(0xc420255900, 0xc400000010, 0x15c1ae0, 0xc42025ba40, 0xb, 0xb, 0x5b, 0x0, 0x0)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/cipher.go:149 +0x80
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*connectionState).readPacket(0xc42000ba00, 0xc42025ba40, 0xc4203764a5, 0x0, 0xb, 0x0, 0x0)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/transport.go:93 +0x68
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*transport).readPacket(0xc42000ba00, 0xc4203764a5, 0xb, 0xb, 0x0, 0xc42003cf38)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/transport.go:89 +0x39
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*handshakeTransport).readOnePacket(0xc4201d1d40, 0xc42025bb60, 0xc42003cf80, 0xb, 0x0, 0x0)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/handshake.go:157 +0x4f
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*handshakeTransport).readLoop(0xc4201d1d40)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/handshake.go:129 +0x2f
created by github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.newClientTransport
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/handshake.go:97 +0x2a6

goroutine 439 [syscall, 9 minutes]:
syscall.Syscall(0x0, 0x7, 0xc4204e7000, 0x1000, 0x456de0, 0xc420415860, 0x4)
	/usr/local/go/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.read(0x7, 0xc4204e7000, 0x1000, 0x1000, 0xc4204f2850, 0x40fe9c, 0xc4204f28a8)
	/usr/local/go/src/syscall/zsyscall_linux_amd64.go:783 +0x55
syscall.Read(0x7, 0xc4204e7000, 0x1000, 0x1000, 0xc4200dbf28, 0x4065bb, 0xc4204f28a8)
	/usr/local/go/src/syscall/syscall_unix.go:161 +0x49
os.(*File).read(0xc4202bc170, 0xc4204e7000, 0x1000, 0x1000, 0x0, 0x0, 0xc4204f2850)
	/usr/local/go/src/os/file_unix.go:228 +0x4f
os.(*File).Read(0xc4202bc170, 0xc4204e7000, 0x1000, 0x1000, 0x70, 0x0, 0x434998)
	/usr/local/go/src/os/file.go:101 +0x59
github.com/vmware/vic/pkg/serial.(*RawConn).Read.func2(0xc4203764b0, 0xc4203764c0, 0xc42025b980, 0xc4204e7000, 0x1000, 0x1000, 0xc4204f29a0)
	/drone/src/github.com/vmware/vic/pkg/serial/rawconn.go:117 +0x55
created by github.com/vmware/vic/pkg/serial.(*RawConn).Read
	/drone/src/github.com/vmware/vic/pkg/serial/rawconn.go:123 +0x152

goroutine 382 [chan receive, 9 minutes]:
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*handshakeTransport).readPacket(0xc42031d560, 0xc4204f1560, 0x1d, 0x1d, 0x0, 0x0)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/handshake.go:120 +0x64
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*mux).onePacket(0xc4204cba40, 0x0, 0x0)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/mux.go:215 +0x37
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*mux).loop(0xc4204cba40)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/mux.go:190 +0x43
created by github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.newMux
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/mux.go:128 +0x1df

goroutine 383 [chan receive, 9 minutes]:
github.com/vmware/vic/cmd/tether.(*attachServerSSH).globalMux(0xc42027a0e0, 0xc42025bf80)
	github.com/vmware/vic/cmd/tether/_test/_obj_test/attach.go:514 +0x15d
created by github.com/vmware/vic/cmd/tether.(*attachServerSSH).run
	github.com/vmware/vic/cmd/tether/_test/_obj_test/attach.go:394 +0x32b

goroutine 384 [chan receive, 9 minutes]:
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*handshakeTransport).readPacket(0xc4201d1d40, 0xc4203764a5, 0xb, 0xb, 0x0, 0x0)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/handshake.go:120 +0x64
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*mux).onePacket(0xc4204cbab0, 0x0, 0x0)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/mux.go:215 +0x37
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*mux).loop(0xc4204cbab0)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/mux.go:190 +0x43
created by github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.newMux
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/mux.go:128 +0x1df

goroutine 385 [chan receive, 9 minutes]:
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*Client).handleGlobalRequests(0xc4204ce9c0, 0xc4204fc120)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/client.go:132 +0x4e
created by github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.NewClient
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/client.go:55 +0xee

goroutine 402 [chan receive, 9 minutes]:
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*Client).handleChannelOpens(0xc4204ce9c0, 0xc4204fc060)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/client.go:141 +0x73
created by github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.NewClient
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/client.go:56 +0x11a

goroutine 403 [semacquire, 9 minutes]:
sync.runtime_notifyListWait(0xc4204ce990, 0x0)
	/usr/local/go/src/runtime/sema.go:267 +0x122
sync.(*Cond).Wait(0xc4204ce980)
	/usr/local/go/src/sync/cond.go:57 +0x80
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*mux).Wait(0xc4204cbab0, 0x0, 0x0)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/mux.go:110 +0x9c
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.NewClient.func1(0xc4204ce9c0)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/client.go:58 +0x33
created by github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.NewClient
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/client.go:60 +0x13c

goroutine 404 [chan receive, 9 minutes]:
github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.(*forwardList).handleChannels(0xc4204ce9d0, 0xc4204fc1e0)
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/tcpip.go:178 +0x71
created by github.com/vmware/vic/vendor/golang.org/x/crypto/ssh.NewClient
	/drone/src/github.com/vmware/vic/vendor/golang.org/x/crypto/ssh/client.go:61 +0x18f

goroutine 409 [semacquire, 9 minutes]:
sync.runtime_notifyListWait(0xc420267410, 0xc400000001)
	/usr/local/go/src/runtime/sema.go:267 +0x122
sync.(*Cond).Wait(0xc420267400)
	/usr/local/go/src/sync/cond.go:57 +0x80
github.com/vmware/vic/pkg/dio.(*multiReader).Read(0xc4202673c0, 0xc420368000, 0x3800, 0x3800, 0x0, 0x0, 0x0)
	/drone/src/github.com/vmware/vic/pkg/dio/reader.go:71 +0x25d
io.copyBuffer(0x7fb6b8d0c278, 0xc4204ccf00, 0x7fb6b8d0c2a0, 0xc4202673c0, 0xc420368000, 0x3800, 0x3800, 0x40f52f, 0xbfd680, 0xc6b2c0)
	/usr/local/go/src/io/io.go:390 +0x147
io.CopyBuffer(0x7fb6b8d0c278, 0xc4204ccf00, 0x7fb6b8d0c2a0, 0xc4202673c0, 0xc420368000, 0x3800, 0x3800, 0x0, 0x0, 0xc4200dd7a8)
	/usr/local/go/src/io/io.go:371 +0x88
github.com/vmware/vic/lib/tether.establishNonPty.func1(0xc42000f8c0)
	/drone/src/github.com/vmware/vic/lib/tether/tether_linux.go:273 +0x12d
created by github.com/vmware/vic/lib/tether.establishNonPty
	/drone/src/github.com/vmware/vic/lib/tether/tether_linux.go:281 +0x36b

goroutine 410 [syscall, 9 minutes]:
syscall.Syscall(0x0, 0x17, 0xc42036c000, 0x3800, 0x29, 0x0, 0x2)
	/usr/local/go/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.read(0x17, 0xc42036c000, 0x3800, 0x3800, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/zsyscall_linux_amd64.go:783 +0x55
syscall.Read(0x17, 0xc42036c000, 0x3800, 0x3800, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/syscall_unix.go:161 +0x49
os.(*File).read(0xc4202bc1a0, 0xc42036c000, 0x3800, 0x3800, 0x0, 0x0, 0x0)
	/usr/local/go/src/os/file_unix.go:228 +0x4f
os.(*File).Read(0xc4202bc1a0, 0xc42036c000, 0x3800, 0x3800, 0x29, 0x0, 0x0)
	/usr/local/go/src/os/file.go:101 +0x59
io.copyBuffer(0x7fb6b8d0c318, 0xc4204dbd40, 0x15c30a0, 0xc4202bc1a0, 0xc42036c000, 0x3800, 0x3800, 0x40f52f, 0xbfd680, 0xd865e0)
	/usr/local/go/src/io/io.go:390 +0x147
io.CopyBuffer(0x7fb6b8d0c318, 0xc4204dbd40, 0x15c30a0, 0xc4202bc1a0, 0xc42036c000, 0x3800, 0x3800, 0x0, 0xc4200dc7a8, 0x56c9f3)
	/usr/local/go/src/io/io.go:371 +0x88
github.com/vmware/vic/lib/tether.establishNonPty.func2(0xc42000f8c0)
	/drone/src/github.com/vmware/vic/lib/tether/tether_linux.go:287 +0x135
created by github.com/vmware/vic/lib/tether.establishNonPty
	/drone/src/github.com/vmware/vic/lib/tether/tether_linux.go:303 +0x2eb

goroutine 411 [syscall, 9 minutes]:
syscall.Syscall(0x0, 0x19, 0xc420370000, 0x3800, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.read(0x19, 0xc420370000, 0x3800, 0x3800, 0x5c7fa0e9f39a3332, 0x40deef, 0x15ec4a0)
	/usr/local/go/src/syscall/zsyscall_linux_amd64.go:783 +0x55
syscall.Read(0x19, 0xc420370000, 0x3800, 0x3800, 0x323895a77213345a, 0x50bb397428bc277e, 0x1)
	/usr/local/go/src/syscall/syscall_unix.go:161 +0x49
os.(*File).read(0xc4202bc1b0, 0xc420370000, 0x3800, 0x3800, 0x10849851c8ca9, 0xc420363e08, 0x40f859)
	/usr/local/go/src/os/file_unix.go:228 +0x4f
os.(*File).Read(0xc4202bc1b0, 0xc420370000, 0x3800, 0x3800, 0x0, 0x40deef, 0x15eb3f8)
	/usr/local/go/src/os/file.go:101 +0x59
io.copyBuffer(0x7fb6b8d0c318, 0xc4204dbd70, 0x15c30a0, 0xc4202bc1b0, 0xc420370000, 0x3800, 0x3800, 0x40f52f, 0xbfd680, 0xd865e0)
	/usr/local/go/src/io/io.go:390 +0x147
io.CopyBuffer(0x7fb6b8d0c318, 0xc4204dbd70, 0x15c30a0, 0xc4202bc1b0, 0xc420370000, 0x3800, 0x3800, 0x9f9827806e98f17e, 0x977e30374772e728, 0x5ca9756b7615972d)
	/usr/local/go/src/io/io.go:371 +0x88
github.com/vmware/vic/lib/tether.establishNonPty.func3(0xc42000f8c0)
	/drone/src/github.com/vmware/vic/lib/tether/tether_linux.go:306 +0x135
created by github.com/vmware/vic/lib/tether.establishNonPty
	/drone/src/github.com/vmware/vic/lib/tether/tether_linux.go:322 +0x310

rax    0xfffffffffffffffc
rbx    0x15e7808
rcx    0x4613e9
rdx    0x80
rdi    0xf
rsi    0x7ffdf5b4ad18
rbp    0x7ffdf5b4b318
rsp    0x7ffdf5b4acd8
r8     0xc42040e0b0
r9     0xc42001f550
r10    0xffffffff
r11    0x246
r12    0x90
r13    0x1b0
r14    0x4
r15    0x19
rip    0x4613e9
rflags 0x246
cs     0x33
fs     0x0
gs     0x0
*** Test killed with quit: ran too long (10m0s).
FAIL	github.com/vmware/vic/cmd/tether	600.013s
make: *** [test-job-github.com/vmware/vic/cmd] Error 1
Makefile:263: recipe for target 'test-job-github.com/vmware/vic/cmd' failed
```

After 910 iterations:
```
ok      github.com/vmware/vic/cmd/tether        5.247s
910
Mar  7 2017 20:56:37.665Z INFO  Started test setup for github.com/vmware/vic/cmd/tether.attachCase
Mar  7 2017 20:56:37.665Z INFO  Using /tmp/tether.attachCase790285343 as test prefix
Mar  7 2017 20:56:37.665Z DEBUG created /tmp/tether.attachCase790285343/ttyS0{c,s} as raw conn pipes
Mar  7 2017 20:56:37.665Z DEBUG created /tmp/tether.attachCase790285343/ttyS1 as persistent log destinations
Mar  7 2017 20:56:37.665Z DEBUG created /tmp/tether.attachCase790285343/ttyS2 as persistent log destinations
Mar  7 2017 20:56:37.669Z DEBUG Test configuration: (func(string, string) error)(0x489e60)
Mar  7 2017 20:56:37.669Z INFO  Registering tether extension mocker
Mar  7 2017 20:56:37.669Z INFO  Registering tether extension Attach
Mar  7 2017 20:56:37.669Z INFO  opening ttyS0 pipe pair for backchannel (client)
Mar  7 2017 20:56:37.669Z DEBUG [BEGIN] [github.com/vmware/vic/lib/tether.(*tether).Start:365] main tether loop
Mar  7 2017 20:56:37.669Z DEBUG [BEGIN] [github.com/vmware/vic/lib/tether.(*tether).setup:111] main tether setup
Mar  7 2017 20:56:38.709Z INFO  Finished test teardown for github.com/vmware/vic/cmd/tether.attachCase
Mar  7 2017 20:56:38.709Z INFO  Started test setup for github.com/vmware/vic/cmd/tether.attachCase
Mar  7 2017 20:56:38.709Z INFO  Using /tmp/tether.attachCase025944306 as test prefix
Mar  7 2017 20:56:38.715Z INFO  Registering tether extension mocker
Mar  7 2017 20:56:38.715Z INFO  Registering tether extension Attach
Mar  7 2017 20:56:38.715Z INFO  opening ttyS0 pipe pair for backchannel (client)
Mar  7 2017 20:56:38.715Z INFO  Trying to establish a connection
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x4886de]

goroutine 108 [running]:
panic(0xbd9780, 0xc42000c0c0)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/vmware/vic/cmd/tether.backchannel.func1(0x15cb020, 0xc42010a940, 0x0, 0x0)
        /home/vagrant/vicsmb/src/github.com/vmware/vic/cmd/tether/attach.go:251 +0x5e
created by github.com/vmware/vic/cmd/tether.backchannel
        /home/vagrant/vicsmb/src/github.com/vmware/vic/cmd/tether/attach.go:253 +0x10b
FAIL    github.com/vmware/vic/cmd/tether        1.071s
```

It looks like we're running two different attachCase tests at the case same time, given multiple instances of `Using /tmp/tether.attachCase<snip> as test prefix` in the output. Don't know why.

The test files contained no useful information for this failure and the files for the other instance of attachCase are cleaned up:
```
vagrant@devbox:/tmp/tether.attachCase025944306$ ls -lrt
total 0
lrwxrwxrwx 1 vagrant vagrant 12 Mar  7 20:56 urandom -> /dev/urandom
-rw-rw-r-- 1 vagrant vagrant  0 Mar  7 20:56 ttyS2
-rw-rw-r-- 1 vagrant vagrant  0 Mar  7 20:56 ttyS1
prwxrwxr-x 1 vagrant vagrant  0 Mar  7 20:56 ttyS0s
prwxrwxr-x 1 vagrant vagrant  0 Mar  7 20:56 ttyS0c
vagrant@devbox:/tmp/tether.attachCase025944306$
```

https://ci.vcna.io/vmware/vic/10995  (without any build parallelization):
```panic: runtime error: invalid memory address or nil pointer dereference
510s
8184
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x8634de]
510s
8186
goroutine 1429 [running]:
510s
8187
github.com/vmware/vic/cmd/tether.backchannel.func1(0xdc6460, 0xc42034d080, 0x0, 0x0)
510s
8188
	github.com/vmware/vic/cmd/tether/_test/_obj_test/attach.go:283 +0x7e
510s
8189
created by github.com/vmware/vic/cmd/tether.backchannel
510s
8190
	github.com/vmware/vic/cmd/tether/_test/_obj_test/attach.go:285 +0x109
510s
8191
FAIL	github.com/vmware/vic/cmd/tether	16.350s
510s
8192
Makefile:261: recipe for target 'test-job-github.com/vmware/vic/cmd' failed```

--- a/cmd/tether/attach.go
+++ b/cmd/tether/attach.go
@@ -251,11 +251,10 @@ func (t *attachServerSSH) stop() error {
 	// This context is used by backchannel only. We need to cancel it before
 	// trying to obtain the following lock so that backchannel interrupts the
 	// underlying Read call by calling Close on it.
-	// The lock is  held by backchannel's caller and not released until it returns
+	// The lock is held by backchannel's caller and not released until it returns
 	log.Debugf("Canceling AttachServer's context")
 	t.cancel()
 
-	log.Debugf("Acquiring the connection lock")
 	t.conn.Lock()
 	if t.conn.conn != nil {
 		log.Debugf("Close called again on rawconn - squashing")
@@ -263,13 +262,15 @@ func (t *attachServerSSH) stop() error {
 		t.conn.conn = nil
 	}
 	t.conn.Unlock()
-	log.Debugf("Released the connection lock")
 
 	return nil
 }
 
 func backchannel(ctx context.Context, conn net.Conn) error {
 	defer trace.End(trace.Begin("establish tether backchannel"))
+
+	// used for shutting down the goroutine cleanly otherwise we leak a goroutine for every successful return from this function
+	done := make(chan struct{})
 
 	// HACK: currently RawConn dosn't implement timeout so throttle the spinning
 	// it does implement the Timeout methods so the intermediary code can be written
@@ -290,6 +291,8 @@ func backchannel(ctx context.Context, conn net.Conn) error {
 		select {
 		case <-ctx.Done():
 			conn.Close()
+		case <-done:
+			return
 		}
 	}()
 
@@ -307,6 +310,7 @@ func backchannel(ctx context.Context, conn net.Conn) error {
 			err := serial.HandshakeServer(conn)
 			if err == nil {
 				conn.SetReadDeadline(time.Time{})
+				close(done)
 				return nil
 			}
 
@@ -367,7 +371,6 @@ func (t *attachServerSSH) run() error {
 					log.Error(detail)
 					return detail
 				}
-
 				// create the SSH server using underlying t.conn
 				t.serverConn.Lock()
 				defer t.serverConn.Unlock()

